lib/pbench/server/database/database.py

```diff
@@ -14,15 +14,16 @@ def get_engine_uri(config, logger):
             psql = config.get("Postgres", "db_uri")
             return psql
         except (NoSectionError, NoOptionError):
-            logger.error(
-                "Failed to find an [Postgres] section" " in configuration file."
-            )
+            if logger:
+                logger.error(
+                    "Failed to find a [Postgres] section in configuration file."
+                )
```

Comment:
This is a bit of a hack; with a logger enabled, `pbench-report-status` was creating empty log files, which caused a lot of unnecessary gold file churn. I hacked the code up to allow initializing the DB without a logger just to keep it quiet.

Reply:
For a user command, logging to `stderr` (if no other logger was set up) would be good (and, probably better than no logging at all...).

Reply:
Yeah; this isn't a "user command"; it's a bridge command for bash scripts to reach the Python ES code to write server report documents, and those bash scripts run in "server" cron jobs. If we have a working server in the first place, our `crontab` configuration will ensure that we have a proper config file, or all this is pointless. So it didn't seem worth the effort to deal with the special case here.